srtcore/core.cpp

```diff
@@ -2835,6 +2835,9 @@ bool srt::CUDT::interpretSrtHandshake(const CHandShake& hs,
                 char target[CSrtConfig::MAX_PFILTER_LENGTH + 1];
                 memset((target), 0, CSrtConfig::MAX_PFILTER_LENGTH + 1);
                 memcpy((target), begin + 1, bytelen);
+                // Un-swap on big endian machines
+                ItoHLA((uint32_t *)target, (uint32_t *)target, blocklen);
```

Comment:
Can `ItoHLA` just replace the above `memcpy` to not copy twice?

Reply:
That seems to work as well, this can be done for connector's requested SID and congestion-control type as well. I'll include those changes in my next PR